python/cinn/compiler/compute_code_generator.py

```diff
@@ -0,0 +1,190 @@
+                ]
+                llir_value = ir._Tensor_.make(
+                    arg_name,
+                    self.inputs_signature[i].dtype,
+                    tensor_shape,
+                    tensor_shape,
+                )
+            else:
+                llir_value = ir.Var(arg_name)
+                llir_args.append(ir.Argument(llir_value))
+                llir_value = ir.Expr(llir_value)
+            self.set_value(arg_name, llir_value)
+
+        # 2. Construct body of function
+        stmts = self.visit_compound_statement(node.body)
+        body = ir.Block.make(stmts)
+
+        # 3. Construct LoweredFunc
+        self.cinn_llir_func = ir.LoweredFunc.make(
+            self.function_name, llir_args, body
+        )
+
+    def visit_compound_statement(self, stmts):
+        cinn_stmts = []
+        for stmt in stmts:
+            cinn_stmt = self.visit(stmt)
+            assert cinn_stmt is not None, f"Unsupport parse:\n{stmt}"
+            if cinn_stmt is None:
+                continue
+            cinn_stmts.append(self.visit(stmt))
+        return cinn_stmts
+
+    def visit_arguments(self, node):
+        arg_names = []
+        for arg in node.args:
+            arg_names += [self.visit(arg)]
+        return arg_names
+
+    def visit_arg(self, node):
+        ast.NodeVisitor.generic_visit(self, node)
+        return node.arg
+
+    def visit_For(self, node) -> ir.Expr:
+        """
+        parse CINN Low Level IR For.
+
+        Args:
+            node(ast.For): The ast For node
+
+        Returns:
+            ir.Expr, Points to the Expr of ir::ExprNode<For>
+        """
+        iter_args = [self.visit(arg) for arg in node.iter.args]
+        assert (
+            len(iter_args) <= 2
+        ), "CINN Low Level IR does not support setting the range step"
+        ast_min = iter_args[0] if len(iter_args) > 1 else self.visit(ast.Num(0))
+        ast_extent = iter_args[1] if len(iter_args) > 1 else iter_args[0]
+
+        # TODO(6clc): support sub region's local variable
+        llir_var = ir.Var(node.target.id)
+        llir_var_expr = ir.Expr(llir_var)
+        self.set_value(node.target.id, llir_var_expr)
+
+        llir_for_min = ir.Expr(ast_min)
+        llir_for_extent = ir.Expr(ast_extent)
+        llir_for_body = self.visit_compound_statement(node.body)
+        llir_for_body = ir.Block.make(llir_for_body)
+        for_expr = ir.For.make(
+            llir_var, llir_for_min, llir_for_extent, llir_for_body
+        )
+        return for_expr
+
+    def visit_Name(self, node):
+        if type(node.ctx) == ast.Store:
+            if node.id in self.local_variables:
+                return self.local_variables[node.id]
+            return node.id
+        # Load Node
+        assert (
+            node.id in self.local_variables
+        ), f"{node.id} is not defined in context"
+        return self.local_variables[node.id]
+
+    def visit_BinOp(self, node):
```

Comment:
Is this a temporary function?

I asked because it only handled Add here, and BinOp can potentially not only Load. So I guess it may not be correct?

Reply:
At present, many cases of `visit_XXX` are not considered clearly.

The code about `visit_XXX` will be re-written by new commit, also in order to support the conversion of cinn llir ast to python ast more easily later.

It is intended to implement the mapping between python ast and llir ast using the Registry schema in next commit.